shared-bindings/busio/SPI.c

```diff
@@ -485,6 +638,14 @@ STATIC const mp_rom_map_elem_t busio_spi_locals_dict_table[] = {
     { MP_ROM_QSTR(MP_QSTR_write), MP_ROM_PTR(&busio_spi_write_obj) },
     { MP_ROM_QSTR(MP_QSTR_write_readinto), MP_ROM_PTR(&busio_spi_write_readinto_obj) },
     { MP_ROM_QSTR(MP_QSTR_frequency), MP_ROM_PTR(&busio_spi_frequency_obj) }
+
+    #if CIRCUITPY_SAMD
```

Comment:
`shared-bindings` shouldn't have conditionalized APIs. Instead, the new module will have the new API and only be implemented on SAMD51.